app/models/booking.rb

```diff
@@ -34,32 +34,67 @@ class Booking < ApplicationRecord
   validates :quantity, numericality:{greater_than: 0}
   validates :item_id, presence: true
   validates :user_id, presence: true
-  validates :start_time, uniqueness: { scope: [:item_id, :user_id, :end_time] }
+  validate :duplicate_booking
   validate :booking_must_be_at_least_one_hour_before
   validate :end_time_must_be_later_than_start_time
   validate :enough_items
 
   def booking_must_be_at_least_one_hour_before
+    return unless self.errors.blank?
+
     if self.start_time < DateTime.now + 1.hours
       self.errors.add(:booking, 'must be done at least one hour before')
     end
   end
 
   def end_time_must_be_later_than_start_time
+    return unless self.errors.blank?
+
     if self.start_time >= self.end_time
       self.errors.add(:end_time, 'must be later than start time')
     end  
   end
 
+  def duplicate_booking
+    return unless self.errors.blank?
+
+    duplicates = Booking.where(item_id: self.item_id, user_id: self.user_id, start_time: self.start_time, end_time: self.end_time, status: 0)
+                        .where.not(id: self.id)
+
+    if duplicates.any?
+      self.errors.add(:base, "Similar pending booking exists, please check")
+    end  
+  end
+
   def enough_items
-    used = Booking.where(item_id: item_id, status: "approved").where('? BETWEEN ? AND ?', self.start_time, start_time, end_time).size
-    available = self.item.quantity - used
+    return unless self.errors.blank?
```

Comment:
What's the purpose of this function enough_items? Is it to check if we have enough items left in inventory?

Reply:
yes